### ISSUE TITLE
toFormat doesn't work in a loop

### ISSUE THREAD

**Issue by aphorism44** [MikeMcl/bignumber.js#107]
I have attached bignumber.js to a phaser.io project, and while the BigNumbers are working fine, the "toFormat" function is not.

Specifically, in the below code block (which is run once every second) , the function works the first time the code runs, and returns the properly-formatted BigNumber:

var format = {
                groupSeparator: ','
                , groupSize: 3
           };
 BigNumber.config({ FORMAT: format });
 return this.moneyPool.toFormat(); //for 1000000, returns "1,000,000" during the first execution

However, the second time the code runs, I get this error on the last line (from Chrome console):

VM828:56 Uncaught TypeError: this.moneyPool.toFormat is not a function
    at Object.GameModel.getMoneyPool (<anonymous>:56:35) //reference to the last line of the code block above
    at Game.timerTrigger (<anonymous>:105:60)
    at Phaser.Timer.update (phaser.js:67390)
    at Phaser.Time.update (phaser.js:66682)
    at Phaser.Game.update (phaser.js:38925)
    at Phaser.RequestAnimationFrame.updateRAF (phaser.js:61994)
    at _onLoop (phaser.js:61978)

This is completely baffling, and I don't know why it keeps happening. I did try to move the configuration code in an init function (so that BigNumber.config only runs once), but the exact same error occurred.

If it matters, I'm not using node; I just use the following line to include your file:

game.load.script('bignumber', 'vendor/bignumber.min.js');

Thanks.

**Comment by MikeMcl** [comment]
If `this.moneyPool.toFormat` is not a function then `this.moneyPool` is probably not a BigNumber.

**Comment by aphorism44** [comment]
Afraid it is. My code is here: https://github.com/aphorism44/townfolkCartel/blob/master/js/GameModel.js. See line 18:

   this.moneyPool = new BigNumber(1000000);

Also, as I mentioned, this.moneyPool.toFormat does run the first time, just not the second.

**Comment by aphorism44** [comment]
Okay, I found this:

When I logged out a typeOf for this.moneyPool, the first pass showed it was an object (correct), but the second pass showed it was a number.

Between the two loops, I do perform "normal" arithmetic on them, such as:

this.moneyPool += this.lastIncome;

Where this.lastIncome is a number, not a BigNumber.

I will _guess_ this means that all operations on a BigNumber must be done with other BigNumbers, and using BigNumber methods (e.g., ".plus" instead of "+"). However, I couldn't find anything in the documentation on this happening, so it is very confusing.
